zcash_client_sqlite/src/scan.rs

```diff
@@ -66,10 +98,12 @@ pub fn scan_cached_blocks<P: AsRef<Path>, Q: AsRef<Path>>(
     let mut last_height = data.query_row("SELECT MAX(height) FROM blocks", NO_PARAMS, |row| {
         row.get(0).or(Ok(SAPLING_ACTIVATION_HEIGHT - 1))
     })?;
-
     // Fetch the CompactBlocks we need to scan
-    let mut stmt_blocks = cache
-        .prepare("SELECT height, data FROM compactblocks WHERE height > ? ORDER BY height ASC")?;
+    let sql = format!(
+        "SELECT height, data FROM compactblocks WHERE height > ? ORDER BY height ASC LIMIT {}",
+        limit
+    );
+    let mut stmt_blocks = cache.prepare(&sql)?;
```

Comment:
Instead of using `format!`, leverage the statement substitution that is built into `rusqlite`:
```rust
    let mut stmt_blocks = cache
        .prepare("SELECT height, data FROM compactblocks WHERE height > ? ORDER BY height ASC LIMIT ?")?;
    let rows = stmt_blocks.query_map(&[last_height, limit], |row| {
```